osf/management/commands/data_storage_usage.py

```diff
@@ -0,0 +1,615 @@
+            start=start,
+            end=end,
+            cursor=cursor,
+        ))
+
+        logger.debug('Gathering supplement summary at {}'.format(datetime.datetime.now()))
+        summary_data = combine_summary_data(summary_data, summarize(
+            sql=ND_PREPRINT_SUPPLEMENT_SIZE_SUM_SQL,
+            content_type=abstractnode_content_type,
+            start=start,
+            end=end,
+            cursor=cursor,
+        ))
+        logger.debug('Gathering deleted file summary at {}'.format(datetime.datetime.now()))
+        cursor.execute(
+            DELETED_FILE_SIZE_SUM_SQL,
+            [
+                start,
+                end,
+            ]
+        )
+        summary_data = combine_summary_data(summary_data, cursor.fetchall())
+        logger.debug('Gathering total file summary at {}'.format(datetime.datetime.now()))
+        cursor.execute(
+            TOTAL_FILE_SIZE_SUM_SQL,
+            [
+                start,
+                end,
+            ]
+        )
+        summary_data = combine_summary_data(summary_data, cursor.fetchall())
+
+        logger.debug('Gathering preprint usage at {}'.format(datetime.datetime.now()))
+        filename = './data-usage-raw-preprints-{}-{}.csv'.format(start, end)
+
+        cursor.execute(
+            PREPRINT_LIST_SQL,
+            [
+                preprint_content_type,
+                preprint_content_type,
+                start,
+                end,
+            ]
+        )
+        if not dry_run:
+            logger.debug('Writing {} to zip.'.format(filename))
+            write_raw_data(cursor=cursor, zip_file=zip_file, filename=filename)
+
+        logger.debug('Gathering preprint summary at {}'.format(datetime.datetime.now()))
+        summary_data = combine_summary_data(summary_data, summarize(
+            sql=ND_PREPRINT_SIZE_SUM_SQL,
+            content_type=preprint_content_type,
+            start=start,
+            end=end,
+            cursor=cursor,
+        ))
+        logger.debug('Gathering regional preprint summary at {}'.format(datetime.datetime.now()))
+        summary_data = combine_summary_data(summary_data, summarize(
+            sql=REGIONAL_PREPRINT_SIZE_SUM_SQL,
+            content_type=preprint_content_type,
+            start=start,
+            end=end,
+            cursor=cursor,
+        ))
+
+    return summary_data
+
+
+def write_summary_data(filename, summary_data, remote_base_folder):
+    header_row = []
+    summary_row = []
+    for key in summary_data:
+        header_row += [key]
+        summary_row += [summary_data[key]]
+    file_path = '{}{}'.format(TEMP_FOLDER, filename)
+    old_remote = requests.get(
+        url=remote_base_folder['files'],
+        auth=bearer_token_auth(DS_METRICS_OSF_TOKEN),
+        params={'filter[name]': filename},
+    ).json()
+    try:
+        old_remote_data = old_remote['data'][0]
+        upload = old_remote_data['links']['upload']
```

Comment:
Only in the case of the file already existing. If it doesn't, then the upload link would go to the folder. That being said, in the block, technically it's the same, but I feel like that may change one day. It's kind of weird that the direct API access url is the upload url (since the download url is for actually downloading the file - might be equivalent if I let it jump through some redirects, but I feel like we may be missing a file link here).

So, essentially, from a practical perspective, yes, they are the same in this branch. Philosophically, they're not, though. That seems like a dumb reason to keep it, though, so removed the redundancy.